common/include/pcl/common/impl/centroid.hpp

```diff
@@ -56,27 +56,31 @@ template <typename PointT, typename Scalar> inline unsigned int
 compute3DCentroid (ConstCloudIterator<PointT> &cloud_iterator,
                    Eigen::Matrix<Scalar, 4, 1> &centroid)
 {
-  // Initialize to 0
-  centroid.setZero ();
-
   unsigned cp = 0;
 
+  // Initialize to 0
+  Eigen::Vector4f accu = Eigen::Vector4f::Zero ();
+
   // For each point in the cloud
   // If the data is dense, we don't need to check for NaN
   while (cloud_iterator.isValid ())
   {
     // Check if the point is invalid
     if (pcl::isFinite (*cloud_iterator))
     {
-      centroid[0] += cloud_iterator->x;
-      centroid[1] += cloud_iterator->y;
-      centroid[2] += cloud_iterator->z;
+      accu += cloud_iterator->getVector4fMap ();
       ++cp;
     }
     ++cloud_iterator;
   }
-  centroid /= static_cast<Scalar> (cp);
+  if (cp == 0)
+  {
+    return (cp);
+  }
+  accu /= static_cast<Scalar> (cp);
+  centroid.col (0) = accu.cast<Scalar> ();
   centroid[3] = 1;
+
```

Comment:
```suggestion
  if (cp)
  {
    accu /= static_cast<Scalar> (cp);
    centroid.col (0) = accu.cast<Scalar> ();
    centroid[3] = 1;
  }
```

Reply:
Should we leave `centroid` untouched or set it to zero (in the case where `cp == 0`)?

Reply:
the docstring says that if the return value is 0, `centroid` is not changed. There was some discussion in the other PR about this, and it was decided to follow the docstring.